module/zfs/dmu_redact.c

```diff
@@ -1007,6 +1007,10 @@ dmu_redact_snap(const char *snapname, nvlist_t *redactnvl,
 	objset_t *os;
 	struct redact_thread_arg *args = NULL;
 	redaction_list_t *new_rl = NULL;
+	char *newredactbook;
+
+	newredactbook = kmem_zalloc(sizeof (char) * ZFS_MAX_DATASET_NAME_LEN,
+	    KM_SLEEP);
```

Comment:
There is a `goto out` a few lines above, so I'll need to put it above that (because the `out` label frees the `newredactbook` buffer) and still fix one possible leak at the name length check.

Reply:
Actually there are several gotos... umm. I'll try to think of something better here...